ui/raidboss/data/07-dt/ultimate/futures_rewritten.ts

```diff
@@ -481,13 +681,895 @@ const triggerSet: TriggerSet<Data> = {
+          return;
 
+        if (debuff === 'ice') {
+          data.p3RelativityRoleMap[role].hasIce = true;
+          data.p3RelativityRoleMap.ice = matches.target;
+        } else if (role === 'dps' && debuff === 'shortFire')
+          data.p3RelativityRoleMap.dps.shortFire.push(matches.target);
+        else if (role === 'support' && debuff === 'longFire')
+          data.p3RelativityRoleMap.support.longFire.push(matches.target);
+        else
+          data.p3RelativityRoleMap[role][debuff] = matches.target;
+
+        if (data.me === matches.target)
+          data.p3RelativityDebuff = debuff;
+      },
+    },
+    {
+      id: 'FRU P3 Ultimate Relativity Initial Debuff',
+      type: 'GainsEffect',
+      netRegex: { effectId: ['997', '99E'], capture: false },
+      condition: (data) => data.phase === 'p3-ur' && data.p3RelativityRoleCount === 8,
+      durationSeconds: 8,
+      infoText: (data, _matches, output) => {
+        const role = data.role === 'dps' ? 'dps' : 'support';
+        const debuff = data.p3RelativityDebuff;
+        if (debuff === undefined)
+          return;
+
+        if (debuff === 'ice')
+          return output.debuffSolo!({ debuff: output.ice!() });
+        else if (debuff === 'mediumFire')
+          return output.debuffSolo!({ debuff: output.mediumFire!() });
+        else if (debuff === 'longFire') {
+          if (role === 'dps')
+            return output.debuffSolo!({ debuff: output.longFire!() });
+          const other = data.p3RelativityRoleMap.support.longFire.find((e) => e !== data.me);
+          return output.debuffShared!({
+            debuff: output.longFire!(),
+            other: data.party.member(other),
+          });
+        }
+        if (role === 'support')
+          return output.debuffSolo!({ debuff: output.shortFire!() });
+        const other = data.p3RelativityRoleMap.dps.shortFire.find((e) => e !== data.me);
+        return output.debuffShared!({
+          debuff: output.shortFire!(),
+          other: data.party.member(other),
+        });
+      },
+      outputStrings: {
+        debuffSolo: {
+          en: '${debuff}',
+        },
+        debuffShared: {
+          en: '${debuff} (w/ ${other})',
+        },
+        shortFire: {
+          en: 'Short Fire',
+        },
+        mediumFire: {
+          en: 'Medium Fire',
+        },
+        longFire: {
+          en: 'Long Fire',
+        },
+        ice: {
+          en: 'Ice',
+        },
+      },
+    },
+    {
+      id: 'FRU P3 Ultimate Relativity Stoplight Collect',
+      type: 'AddedCombatant',
+      netRegex: { npcBaseId: '17832' },
+      run: (data, matches) => data.p3RelativityStoplights[matches.id] = matches,
+    },
+    {
+      id: 'FRU P3 Ultimate Relativity Y North Spot',
+      type: 'Tether',
+      // boss tethers to 5 stoplights - 0085 are purple tethers, 0086 are yellow
+      netRegex: { id: '0086' },
+      // condition: (data) => data.triggerSetConfig.ultimateRel === 'yNorthDPSEast',
```

Comment:
Pointless comment? Or accidentally commented out?

Reply:
Testing artifact, but this actually does need a `data.phase` condition since p4 has yellow stoplight tethers too.